typed-racket-lib/typed-racket/private/type-contract.rkt

```diff
@@ -385,6 +394,24 @@
          (if numeric-sc
              (apply or/sc numeric-sc (map t->sc non-numeric))
              (apply or/sc (map t->sc elems)))]
+        [(Intersection: ts)
```

Comment:
I think this code is correct, but it was built by iteratively throwing things at the wall and seeing what stuck. Would appreciate a sanity check here.